src/services/codeFixProvider.ts

```diff
@@ -59,9 +59,10 @@ export function registerCodeFix(reg: CodeFixRegistration) {
     }
 }
 
+let errorCodeToFixesArray: readonly string[] | undefined;
 /** @internal */
 export function getSupportedErrorCodes(): readonly string[] {
-    return arrayFrom(errorCodeToFixes.keys());
+    return errorCodeToFixesArray ??= arrayFrom(errorCodeToFixes.keys());

```

Comment:
Codefixes are registered at runtime, so if this call happens too early it may get locked in forever with a reduced list.

Can we add something like `errorCodeToFixesArray = undefined` to `registerCodeFix` above just to make sure this function is always accurate?

(I want to get rid of this registration mechanism at some point, as it's side-effect based and could be redone post-modules, but, that's another change.)

Reply:
right now how this is setup is that this is immutable list and is correct when called. I will go ahead and add assert in register code fix though

Reply:
You'd be surprised; I know of a few projects which add more on: https://github.com/google/tsec/blob/507e8030de276de3324cac49dfc3abcadacc3a3f/packages/tsec/language_service_plugin.ts#L152

Reply:
wow .. thats just misusing it.. their plugin instead needs to implement `getSupportedCodeFixes` on LS

Reply:
An assert would stop those cases from functioning, right?

Reply:
Yeah. I guess if we have an intended way to do this, then we can ban this unsupported one (this register mechanism may go away and that is all internal).

Reply:
> An assert would stop those cases from functioning, right?

Yeah for now i have just gone ahead with resetting the list.

Reply:
To be clear I'm not against breaking this particular mechanism of adding code fixes. I'm curious why they did this; hopefully it was just because they didn't know about `getSupportedCodeFixes`, but maybe there's a restriction there I'm not thinking of. tsec is one that I think is actively maintained. The roblox one is... confusing because it doesn't add anything.

Reply:
The restriction is that IDEs don't use the feature so the function is never called and the proxy is useless. I created an issue: https://github.com/microsoft/vscode/issues/219273

Reply:
Isn't making more people depend on this an even worse direction?

Reply:
Which direction?
I don't have much opinion on the design, but I think the language server should have an official API to extend code actions.